esmvaltool/preprocessor/_download.py

```diff
@@ -60,21 +61,40 @@ def synda_download(synda_name, dest_folder):
             'synda', 'get', '--dest_folder={}'.format(dest_folder),
             '--verify_checksum', synda_name
         ]
-        cmd = ' '.join(cmd)
-        logger.debug("Running: %s", cmd)
-        subprocess.check_call(cmd, shell=True)
+        logger.debug("Running: %s", ' '.join(cmd))
+        subprocess.check_call(cmd)
 
     return local_file
 
 
-def download(files, dest_folder):
+def download(files, dest_folder, variable, rootpath, drs,
+             to_repository, compress_downloads):
     """Download files that are not available locally"""
+    if to_repository:
+        dest_folder = list_input_dirs(
+            variable, rootpath, drs, skip_non_existent=False
+        )[0]
+
     if not os.path.exists(dest_folder):
         os.makedirs(dest_folder)
 
     local_files = []
     for name in files:
-        local_file = synda_download(synda_name=name, dest_folder=dest_folder)
+        local_file = synda_download(
+            synda_name=name,
+            dest_folder=dest_folder,
+            variable=variable,
+            rootpath=rootpath,
+            drs=drs
+        )
+        if compress_downloads:
+            cmd = [
+                'nccopy', '-4', '-d', '4', '-s',
```

Comment:
This adds a dependency on a software package that provides the `nccopy` command. What (conda) package provides this, is it already a dependency of esmvaltool?

Reply:
It is part of netcdf installation, no new dependencies needed